agent/taskresource/cgroup.go

```diff
@@ -0,0 +1,225 @@
+func NewCgroupResource(cgroupRoot string, cgroupMountPath string) *CgroupResource {
+	return &CgroupResource{
+		// control creates interface to cgroups library methods
+		control:         cgroup.New(),
+		CgroupRoot:      cgroupRoot,
+		CgroupMountPath: cgroupMountPath,
+		ioutil:          ioutilwrapper.NewIOUtil(),
+	}
+}
+
+// SetDesiredStatus safely sets the desired status of the resource
+func (c *CgroupResource) SetDesiredStatus(status CgroupStatus) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.desiredStatusUnsafe = status
+}
+
+// GetDesiredStatus safely returns the desired status of the task
+func (c *CgroupResource) GetDesiredStatus() CgroupStatus {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.desiredStatusUnsafe
+}
+
+// SetKnownStatus safely sets the currently known status of the resource
+func (c *CgroupResource) SetKnownStatus(status CgroupStatus) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.knownStatusUnsafe = status
+}
+
+// GetKnownStatus safely returns the currently known status of the task
+func (c *CgroupResource) GetKnownStatus() CgroupStatus {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.knownStatusUnsafe
+}
+
+// SetCreatedAt sets the timestamp for resource's creation time
+func (c *CgroupResource) SetCreatedAt(createdAt time.Time) {
+	if createdAt.IsZero() {
+		return
+	}
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.createdAt = createdAt
+}
+
+// GetCreatedAt sets the timestamp for resource's creation time
+func (c *CgroupResource) GetCreatedAt() time.Time {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.createdAt
+}
+
+func (c *CgroupResource) Create(task *api.Task) error {
+	err := c.cgroupInit()
+	if err != nil {
+		seelog.Criticalf("unable to setup '/ecs' cgroup: %v", err)
+		return err
+	}
+	err = c.setupTaskCgroup(task)
+	if err != nil {
+		seelog.Criticalf("unable to setup platform resources: %v", err)
+		return err
+	}
+	return nil
+}
+
+func (c *CgroupResource) cgroupInit() error {
+	if c.control.Exists(config.DefaultTaskCgroupPrefix) {
+		seelog.Debugf("Cgroup at %s already exists, skipping creation", config.DefaultTaskCgroupPrefix)
+		return nil
+	}
+	return c.control.Init()
+}
+
+func (c *CgroupResource) setupTaskCgroup(task *api.Task) error {
+	cgroupRoot := c.CgroupRoot
+	seelog.Debugf("Setting up cgroup at: %s for task: %s", cgroupRoot, task.Arn)
+
+	if c.control.Exists(cgroupRoot) {
+		seelog.Debugf("Cgroup at %s already exists, skipping creation", cgroupRoot)
+		return nil
+	}
+
+	linuxResourceSpec, err := task.BuildLinuxResourceSpec()
+	if err != nil {
+		return errors.Wrapf(err, "resource: setup cgroup: unable to build resource spec for task: %s", task.Arn)
+	}
+
+	cgroupSpec := cgroup.Spec{
+		Root:  cgroupRoot,
+		Specs: &linuxResourceSpec,
+	}
+
+	_, err = c.control.Create(&cgroupSpec)
+	if err != nil {
+		return errors.Wrapf(err, "resource: setup cgroup: unable to create cgroup at %s for task: %s", cgroupRoot, task.Arn)
+	}
+
+	// echo 1 > memory.use_hierarchy
```

Comment:
Can you change this comment to be more descriptive, `enable cgroup memory hierarchy`?

Reply:
Sure

Reply:
Would it be reasonable to include both? I put that comment in originally because it is the command you run on host to get the same effect.